packages/cli/lib/index.ts

```diff
@@ -95,14 +97,33 @@ program
     .command('init')
     .argument('[path]', 'Optional: The path to initialize the Nango project in. Defaults to the current directory.')
     .description('Initialize a new Nango project')
-    .action(function (this: Command) {
-        const { debug } = this.opts<GlobalOptions>();
-        const absolutePath = path.resolve(process.cwd(), this.args[0] || '');
-        const ok = init({ absolutePath, debug });
+    .action(async function (this: Command) {
+        const { debug, zero } = this.opts<GlobalOptions>();
+        const currentPath = process.cwd();
+        const absolutePath = path.resolve(currentPath, this.args[0] || '');
 
-        if (ok) {
-            console.log(chalk.green(`Nango integrations initialized in ${absolutePath}!`));
+        const check = await verificationService.preCheck({ fullPath: absolutePath, debug });
+        if (check.hasNangoYaml || check.isZeroYaml) {
+            console.log(chalk.red(`The path provided is already a Nango integrations folder.`));
+            return;
+        }
+
+        if (zero) {
+            const res = await initZero({ absolutePath, debug });
+            if (!res) {
+                process.exitCode = 1;
```

Comment:
should we show an error message?

Reply:
ideally I would keep the printing out of the internal functions and only in the most outer layer but it is not always easy I understand